### PR TITLE
[release/9.0] Fixed AddDockerFile to work with compute customization

### PR DESCRIPTION
**Pull request by davidfowl**
Backport of #6442 to release/9.0

## Customer Impact

Using docker files with container app customization does not work currently. 

## Testing

Unit tests, and manual testing with azd.

## Risk

Low.

## Regression?

No
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/6446)

### CODE REVIEW

File: playground/AzureContainerApps/AzureContainerApps.AppHost/pythonapp.module.bicep [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          server: outputs_azure_container_registry_endpoint
+          identity: outputs_azure_container_registry_managed_identity_id
+        }
+      ]
+    }
+    environmentId: outputs_azure_container_apps_environment_id
+    template: {
+      containers: [
+        {
+          image: pythonapp_containerimage
```
Comment (davidfowl) [inline playground/AzureContainerApps/AzureContainerApps.AppHost/pythonapp.module.bicep:31]: This is the key change.

**Review by mitchdenny — APPROVED** [review]

File: tests/Aspire.Hosting.Azure.Tests/AzureContainerAppsTests.cs [hunk 2]
```diff
@@ -104,6 +104,114 @@ param outputs_azure_container_apps_environment_id string
...
+          "path": "api.module.bicep",
+          "params": {
+            "outputs_azure_container_registry_managed_identity_id": "{.outputs.AZURE_CONTAINER_REGISTRY_MANAGED_IDENTITY_ID}",
+            "outputs_managed_identity_client_id": "{.outputs.MANAGED_IDENTITY_CLIENT_ID}",
+            "outputs_azure_container_apps_environment_id": "{.outputs.AZURE_CONTAINER_APPS_ENVIRONMENT_ID}",
+            "outputs_azure_container_registry_endpoint": "{.outputs.AZURE_CONTAINER_REGISTRY_ENDPOINT}",
+            "api_containerimage": "{api.containerImage}"
+          }
+        }
+        """;
```
Comment (eerhardt) [inline tests/Aspire.Hosting.Azure.Tests/AzureContainerAppsTests.cs:152]: Should we be baselining the entire resource's manifest? And not just the `deployment` section?

**Review by eerhardt — APPROVED** [review]

File: tests/Aspire.Hosting.Azure.Tests/AzureContainerAppsTests.cs (hunk 2 above)
Comment (davidfowl) [reply tests/Aspire.Hosting.Azure.Tests/AzureContainerAppsTests.cs:152]: We could but it's not really relevant for this test.

Comment (eerhardt) [reply tests/Aspire.Hosting.Azure.Tests/AzureContainerAppsTests.cs:152]:
```json
    "pythonapp": {
      "type": "container.v1",
      "build": {
        "context": "AppWithDocker",
        "dockerfile": "AppWithDocker/Dockerfile"
      },
```

I don't understand how that part isn't relevant. Seems pretty important for the scenario to work.

**Pull request merged**